cpp/src/arrow/json/parser.cc

```diff
@@ -102,7 +104,7 @@ Status Kind::ForType(const DataType& type, Kind::type* kind) {
     Status Visit(const BinaryType&) { return SetKind(Kind::kString); }
     Status Visit(const LargeBinaryType&) { return SetKind(Kind::kString); }
     Status Visit(const TimestampType&) { return SetKind(Kind::kString); }
-    Status Visit(const FixedSizeBinaryType&) { return SetKind(Kind::kString); }
+    Status Visit(const FixedSizeBinaryType&) { return SetKind(Kind::kNumberOrString); }
```

Comment:
Hmm, that doesn't seem right to me. Ideally we would only allow numbers for decimal, not for all fixed-size binary types.

Reply:
changed to Decimal